examples/voice-pipeline-agent/function_calling_weather.py

```diff
@@ -33,6 +33,18 @@ async def get_weather(
         ],
     ):
         """Called when the user asks about the weather. This function will return the weather for the given location."""
+
+        # Example of a filler message while waiting for the function call to complete.
+        # NOTE: This message illustrates how the agent can engage users by using the `say()` method
+        # while awaiting the completion of the function call. To create a more dynamic and engaging
+        # interaction, consider varying the responses based on context or user input.
+        call_ctx = AgentCallContext.get_current()
+        message = f"Let me check the weather in {location} for you."
+        await call_ctx.agent.say(message)
+
+        # (optional) add the filler message to the chat context for synthesis the tool call speech
+        call_ctx.chat_ctx.append(text=message, role="assistant")
```

Comment:
Can we allow users to manage the chat ctx used for tool call answer synthesis? For example, if a user inserts some messages using "say" in a function call, they can choose to attach that messages to the chat ctx used to synthesize the answer later.

This would allow us to trigger the llm at the end of the function call instead of waiting for the speech to finish and using the latest global chat ctx. See the difference in this commit https://github.com/livekit/agents/pull/1130/commits/d24c43766b9f460b7638b869e0493801932de94d.

@theomonnom please let me know what do you think.

Reply:
Yes, I think it is a great idea! Users should be able to edit the chat_ctx inside the call_ctx

Reply:
`await call_ctx.agent.say(message, add_to_chat_ctx=True)` should be automatic.

There should be no need to explicitly call `call_ctx.chat_ctx.append(text=message, role="assistant")` if you used the `say` method


But for other usecases like injecting RAG for the current call, I think `call_ctx.chat_ctx.append` makes a ton of sense.

Reply:
`await call_ctx.agent.say(message, add_to_chat_ctx=True)` only updates the chat ctx after speech playing so the answer synthesis of the function call needs to wait for the speech finished. By manually updating `call_ctx.chat_ctx`, we can trigger a llm call once the function completes, and the user can decide what to put into the chat ctx.

Reply:
Right I see, I think we should maybe add a new argument to `agent.say`? To declare if we want to wait or not?
Maybe we should make `agent.say` sync but it returns an `asyncio.Future` that users can optionally choose to wait?

wdyt?

Reply:
Let `say` return a `Future` do makes sense in case some one want to wait for the speech, but I don't think waiting for `say` in a function call can solve the problem. It's still decided by the pipeline agent where to synthesize the tool call answer,
1. right after the function call returned 
2. or until this tool call speech handle being processed in `play_speech`

Perhaps we can make this an option to the user from the function call?

Reply:
We have to wait for the function call to returns. Because we use the return value as the "answer"

Reply:
Yeah, both of the options will wait for the function call returned. I'll send you a message in slack.